cmd/subnetcmd/publish.go

```diff
@@ -0,0 +1,380 @@
+			choice, err := app.Prompt.CaptureList("Don't know which repo to publish to. How would you like to proceed?", options)
+			if err != nil {
+				return err
+			}
+			if choice == options[0] {
+				// user chose to provide a new alias
+				alias, err = getNewAlias()
+				if err != nil {
+					return err
+				}
+				// double-check: actually this path exists...
+				if _, err := os.Stat(filepath.Join(reposDir, alias)); err == nil {
+					ux.Logger.PrintToUser("The repository with the given alias already exists locally. You may have already published this subnet there (the other explanation is that a different subnet has been published there).")
+					yes, err := app.Prompt.CaptureYesNo("Do you wish to continue?")
+					if err != nil {
+						return err
+					}
+					if !yes {
+						ux.Logger.PrintToUser("User canceled, nothing got published.")
+						return nil
+					}
+				}
+			} else {
+				aliases := make([]string, len(matches))
+				for i, a := range matches {
+					aliases[i] = a.Name()
+				}
+				alias, err = app.Prompt.CaptureList("Pick an alias", aliases)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// ask for a new alias
+func getNewAlias() (string, error) {
+	return app.Prompt.CaptureString("Please provide an alias for the repository we are going to use")
+}
+
+func getRepoURL(reposDir string) error {
+	if repoURL == "" {
+		path := filepath.Join(reposDir, alias)
+		repo, err := git.PlainOpen(path)
+		if err != nil {
+			// TODO: probably a debug log; the alias might not have been created yet,
+			// so the repo might not exist, therefore just ignore it!
+		} else {
+			// there is a repo already for this alias, let's try to figure out the remote URL from there
+			conf, err := repo.Config()
+			if err != nil {
+				// TODO Would we really want to abort here?
+				return err
+			}
+			remotes := make([]string, len(conf.Remotes))
+			i := 0
+			for _, r := range conf.Remotes {
+				// TODO: Usually there's only one URL for a remote, but more are supported - should we too?
+				remotes[i] = r.URLs[0]
+				i++
+			}
+			repoURL, err = app.Prompt.CaptureList("Which is the remote URL for this repo?", remotes)
+			if err != nil {
+				// TODO Would we really want to abort here?
+				return err
+			}
+			return nil
+		}
+		repoURL, err = app.Prompt.CaptureString("Please provide the repository URL")
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// loadYAMLFile loads a YAML file from disk into a concrete types.Definition object
+// using generics. It's role really is solely to verify that the YAML content is valid.
+func loadYAMLFile[T types.Definition](path string, defType T) error {
+	fileBytes, err := os.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	return yaml.Unmarshal(fileBytes, &defType)
+}
+
+func getSubnetInfo(sc *models.Sidecar) (*types.Subnet, error) {
+	homepage, err := app.Prompt.CaptureEmpty("What is the homepage of the Subnet project?", nil)
```

Comment:
CaptureString. Also for the rest

Reply:
CaptureString requires a valid string. But according to @dasconnor we want to allow people to be able to just hit enter everywhere - which means allow empty values.